core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -186,77 +160,70 @@ class KafkaApis(val requestChannel: RequestChannel,
       trace(s"Handling request:${request.requestDesc(true)} from connection ${request.context.connectionId};" +
         s"securityProtocol:${request.context.securityProtocol},principal:${request.context.principal}")
 
-      val handled = request.envelope.exists { envelope =>
```

Comment:
We don't need to check ```envelope``` for each request type now?

Reply:
We do all the same validation in `handleEnvelope` now. Previously the parsing would get done in SocketServer, which means that when the embedded request is first handled at this layer, the envelope still needs authorization. Now we will not even create the embedded request object unless the envelope has been authorized.